### PR TITLE
Fix bug composing parameters for rest url

### PR DESCRIPTION
**Pull request by Nslaver**
The current code is assuming that both arrays are on the same order, resulting in a pagination bug when the parameter array is composed.

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | There must be a single space before the last parenthesis | [Archive_Event.php:301](https://github.com/moderntribe/the-events-calendar/blob/e0266b479e1d9610692c989816601454d998360b/src/Tribe/REST/V1/Endpoints/Archive_Event.php#L301)
:x: | There must be a single space before the last parenthesis | [Archive_Event.php:302](https://github.com/moderntribe/the-events-calendar/blob/e0266b479e1d9610692c989816601454d998360b/src/Tribe/REST/V1/Endpoints/Archive_Event.php#L302)
:x: | No space after opening parenthesis is prohibited | [Archive_Event.php:302](https://github.com/moderntribe/the-events-calendar/blob/e0266b479e1d9610692c989816601454d998360b/src/Tribe/REST/V1/Endpoints/Archive_Event.php#L302)
:x: | There must be a single space after the first parenthesis | [Archive_Event.php:302](https://github.com/moderntribe/the-events-calendar/blob/e0266b479e1d9610692c989816601454d998360b/src/Tribe/REST/V1/Endpoints/Archive_Event.php#L302)
:x: | There must be a single space after the first parenthesis | [Archive_Event.php:302](https://github.com/moderntribe/the-events-calendar/blob/e0266b479e1d9610692c989816601454d998360b/src/Tribe/REST/V1/Endpoints/Archive_Event.php#L302)
:x: | There must be a single space before the last parenthesis | [Archive_Event.php:302](https://github.com/moderntribe/the-events-calendar/blob/e0266b479e1d9610692c989816601454d998360b/src/Tribe/REST/V1/Endpoints/Archive_Event.php#L302)
:x: | No space before closing parenthesis is prohibited | [Archive_Event.php:302](https://github.com/moderntribe/the-events-calendar/blob/e0266b479e1d9610692c989816601454d998360b/src/Tribe/REST/V1/Endpoints/Archive_Event.php#L302)
:x: | There must be a single space after the first square bracket | [Archive_Event.php:303](https://github.com/moderntribe/the-events-calendar/blob/e0266b479e1d9610692c989816601454d998360b/src/Tribe/REST/V1/Endpoints/Archive_Event.php#L303)
:x: | There must be a single space before the last square bracket | [Archive_Event.php:303](https://github.com/moderntribe/the-events-calendar/blob/e0266b479e1d9610692c989816601454d998360b/src/Tribe/REST/V1/Endpoints/Archive_Event.php#L303)
:x: | There must be a single space after the first square bracket | [Archive_Event.php:303](https://github.com/moderntribe/the-events-calendar/blob/e0266b479e1d9610692c989816601454d998360b/src/Tribe/REST/V1/Endpoints/Archive_Event.php#L303)
:x: | There must be a single space before the last square bracket | [Archive_Event.php:303](https://github.com/moderntribe/the-events-calendar/blob/e0266b479e1d9610692c989816601454d998360b/src/Tribe/REST/V1/Endpoints/Archive_Event.php#L303)
:x: | Space found before comma in function call | [Archive_Event.php:305](https://github.com/moderntribe/the-events-calendar/blob/e0266b479e1d9610692c989816601454d998360b/src/Tribe/REST/V1/Endpoints/Archive_Event.php#L305)

> via jenkins/codesniffer-phpcs

File: src/Tribe/REST/V1/Endpoints/Archive_Event.php [hunk 1]
```diff
@@ -298,7 +298,11 @@ protected function get_current_rest_url( array $args = array() ) {
 		$keys    = array_intersect_key( $flipped, $values );
 
 		if ( ! empty( $keys ) ) {
-			$url = add_query_arg( array_combine( array_values( $keys ), array_values( $values ) ), $url );
+			$param = array_fill_keys( array_values( $keys ), '' );
```
Comment (lucatume) [inline src/Tribe/REST/V1/Endpoints/Archive_Event.php]: Nitpicking here: for the sake of clarity could you rename this from `$param` to `$paremeters`?

**Comment by lucatume** [comment]
Hi there @Nslaver,

thanks for opening a Pull Request and providing the fix.
As you can see I've reported your code in this PR: https://github.com/moderntribe/the-events-calendar/pull/1581; the code you fixed has been moved to a parent base class and that is the reason for it: your code works as intended and should only require a small variable renaming.

**Comment by Nslaver** [comment]
@lucatume That's great thanks for the feedback; I saw that you are exposing new functionality at bucket/full-rest-api branch, is there a planned release date for this new features?

**Comment by lucatume** [comment]
The release is planned for "as soon as possible"; I know how that sounds but if, as I think, you can read the code you will understand the state of things.

**Comment by lucatume** [comment]
Ticketed at https://central.tri.be/issues/88748

**Pull request merged**